### PR TITLE
Use Nan package to offer v0.11 and v0.10 support

### PR DESCRIPTION
**Pull request by parkr**
Using this project to abstract away the nastiness of the upgrade: https://github.com/rvagg/nan

This PR should work with both v0.11 AND v0.10. Yay!

Replaces #9.

/cc @FooBarWidget @wanderer @vladlosev @MrNice

### CODE REVIEW

File: src/addon.cpp [hunk 1]
```diff
@@ -26,49 +27,58 @@ class SHA3Hash: public ObjectWrap {
 
 	static void
 	Initialize(Handle<Object> target) {
-		HandleScope scope;
-		Local<FunctionTemplate> t = FunctionTemplate::New(New);
+		NanScope();
```
Comment (vladlosev) [inline src/addon.cpp]: Invoking a constructor in this manner creates a temporary object which is destroyed at the end of the statement. This is essentially a no-op. Give the new object a name: `NanScope scope`. The destructor will then be invoked a the end of the enclosing block, as intended.

Comment (parkr) [inline src/addon.cpp]: @vladlosev It's a macro I think. The example code (throughout that entire file) doesn't ever do that: https://github.com/rvagg/nan#api_nan_scope

Comment (vladlosev) [inline src/addon.cpp]: Camel case macros. Eeew.

**Comment by parkr** [comment]
Travis looks good: https://travis-ci.org/parkr/node-sha3/builds/47591639

**Comment by FooBarWidget** [comment]
This for the pull request, but I'm not maintaining this repository at this moment. Would you be interested in taking over maintenance?

**Comment by parkr** [comment]
@FooBarWidget: sure. Can you please give parkrmoore@gmail.com access to the npm package as well so I can publish the new version? Thanks.

**Comment by FooBarWidget** [comment]
You now have write access to the git repo. But how do I give you access to the npm package? I can't find the interface on npmjs.org for doing that.

**Comment by parkr** [comment]
@FooBarWidget Run this from the command-line:

```bash
$ npm owner add parkr sha3
```

:smiley_cat: 

I don't have a merge button so I don't have commit access yet.

**Comment by parkr** [comment]
@FooBarWidget Can you please enable TravisCI? https://travis-ci.org/profile/phusion

I want to have the tests enabled so we all know this code works. :) Thank you!

**Comment by FooBarWidget** [comment]
Done and done.

**Comment by parkr** [comment]
Thank you, @FooBarWidget!

**Pull request merged**